File: coco/src/main/java/coco/ep/m6/Divd.java
package coco.ep.m6;

public interface Divd<FT> extends coco.ep.m5.Divd<FT>, Exp<FT> {

    default Boolean equals(coco.ep.Exp<FT> other) {
        return convert(this).astree().equals(convert(other).astree());
    }

    default Boolean isDivd(coco.ep.Exp<FT> left, coco.ep.Exp<FT> right) {
        return convert(left).eql(getLeft()) && convert(right).eql(getRight());
    }

    default Boolean eql(coco.ep.Exp<FT> that) {
        return convert(that).isDivd(convert(getLeft()), convert(getRight()));
    }
}

File: coco_csharp/CoCo/EP/M6/Divd.cs
namespace EP.M6 {
    public interface Divd<FT> : Exp<FT>, EP.M5.Divd<FT> {
        bool Exp<FT>.equals(EP.Exp<FT> other) {
            return convert(this).astree().Equals(convert(other).astree());
        }

        bool Exp<FT>.isDivd(EP.Exp<FT> left, EP.Exp<FT> right) {
            return convert(left).eql(getLeft()) && convert(right).eql(getRight());
        }

        bool Exp<FT>.eql(EP.Exp<FT> that) {
            return convert(that).isDivd(convert(getLeft()), convert(getRight()));
        }
    }
}